src/lib/metrics/MetricReport.ts

```diff
@@ -0,0 +1,34 @@
+import { SessionManager } from '../session/SessionManager'
+import * as pkg from '../../../package.json'
+
+export interface Metric {
+  [metricName: string]: number | string | MetricEnv
+}
+
+interface MetricEnv {
+  command: string
+  account: string
+  workspace: string
+  toolbeltVersion: string
+  nodeVersion: string
+  platform: string
+}
+
+export interface MetricReport extends Metric {
+  env: MetricEnv
+}
+
+export function metricToMetricReport(metric: Metric): MetricReport {
+  const { workspace, account } = SessionManager.getSessionManager()
+  return {
+    ...metric,
+    env: {
+      command: process.argv.slice(2).join(' '),
```

Comment:
I think there may be many corner cases to this, for example:
```
vtex --verbose link
```
This will catch all flags on the command as well, giving us too much granular visibility.
We could discover how to get the command, then this could store values like `workspace:reset`, `link`, ...

Reply:
For discovering how to get the command name we could take a look on the code https://github.com/vtex/findhelp

Reply:
True, I did't think about this case!